pkg/ottl/functions.go

```diff
@@ -203,18 +218,36 @@ func (p *basePath[K]) isComplete() error {
 	return p.nextPath.isComplete()
 }
 
-func newKeys[K any](keys []key) []Key[K] {
+func (p *Parser[K]) newKeys(keys []key) ([]Key[K], error) {
 	if len(keys) == 0 {
-		return nil
+		return nil, nil
 	}
 	ks := make([]Key[K], len(keys))
 	for i := range keys {
+		var par Getter[K]
+		if keys[i].Expression != nil {
+			if keys[i].Expression.Path != nil {
+				arg, err := p.buildGetSetterFromPath(keys[i].Expression.Path)
+				if err != nil {
+					return nil, err
+				}
+				par = arg
+			}
+			if keys[i].Expression.Converter != nil {
+				g, err := p.newGetterFromConverter(*keys[i].Expression.Converter)
+				if err != nil {
+					return nil, err
+				}
+				par = g
+			}
+		}
 		ks[i] = &baseKey[K]{
 			s: keys[i].String,
 			i: keys[i].Int,
+			g: par,
```

Comment:
```suggestion
		var getter Getter[K]
		if keys[i].Expression != nil {
			if keys[i].Expression.Path != nil {
				arg, err := p.buildGetSetterFromPath(keys[i].Expression.Path)
				if err != nil {
					return nil, err
				}
				getter = arg
			}
			if keys[i].Expression.Converter != nil {
				g, err := p.newGetterFromConverter(*keys[i].Expression.Converter)
				if err != nil {
					return nil, err
				}
				getter = g
			}
		}
		ks[i] = &baseKey[K]{
			s: keys[i].String,
			i: keys[i].Int,
			g: getter,
```
Minor, just a variable rename suggestion.